waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -526,17 +535,41 @@ proc resume*(ws: WakuStore, peerList: Option[seq[PeerInfo]] = none(seq[PeerInfo]
   lastSeenTime = max(lastSeenTime - offset, 0)
   debug "the  offline time window is", lastSeenTime=lastSeenTime, currentTime=currentTime
 
+  var dismissed: uint = 0
+  var added: uint = 0
+
   proc handler(response: HistoryResponse) {.gcsafe.} =
+    debug "resume handler is called"
+
+    # exclude index from the comparison criteria
+    let currentMsgSummary = ws.messages.map(proc(x: IndexedWakuMessage): WakuMessage = x.msg)
```

Comment:
I'm wondering if there is a way to do this more effectively. I don't think this is the only place where we essentially create a copy of the existing history - perhaps we could instead maintain a single, ordered hash set of messages in the store? This may also speed up `isDuplicate` - finding a match should be more effective on a `HashSet` than on a `seq`

Reply:
Thanks for the comment! Agree, the hash set would work here and indeed removes the duplicates by default (no need to even check). Nevertheless, I expect that the hash set should be slower than the normal seq in terms of insertion (which is the most recurring operation in the store protocol) as it involves hashing. Also, wrt storage, I think hash sets consume more space than sequences (depending on the HashSet implementation).  Let me investigate it a bit further and make sure there would not be a huge performance issue with the hash set compared to seq, and then I will update it.

Reply:
Here is the link to a discussion in Nimbus help-desk about hashset vs seq: https://discord.com/channels/613988663034118151/616299964242460682/855126317498105887
Following this conversation, either we can do a benchmarking on HashSet and seq and pick the one that performs the best, or we can proceed with replacing seq with HashSet with no stress test. There is a third option of keeping it as seq. 
@oskarth @jm-clius Let me know your thoughts.

Reply:
If I understand the thread of conversation, moving to `HashSet` will require at least one less copy of the historical messages? It also naturally deduplicates which will simplify what we have, so if it's a relatively easy change I'd go with "replacing seq with HashSet with no stress test". We still have plans to do proper benchmarking of all our protocols, so this could be a future task.
I'm also fine with the change to HashSet being deferred to a future PR.

Reply:
> We still have plans to do proper benchmarking of all our protocols, so this could be a future task

Oh, great! Then I will replace it with HashSet, though I will do it in a separate PR, as it has a large surface of changes namely, the pagination, DB handlers, the history query internals, and many more.